projects/rocprofiler-systems/tests/rocprof-sys-causal-tests.cmake

```
@@ -35,6 +35,28 @@ rocprofiler_systems_add_causal_test(
         "Starting causal experiment #1(.*)causal/experiments.json(.*)causal/experiments.coz"
 )
 
+rocprofiler_systems_add_causal_test(
+    SKIP_BASELINE
+    NAME cpu-rocprofsys-func-verbose
+    TARGET causal-cpu-rocprofsys
+    RUN_ARGS 70 10 432525 400000000
+    CAUSAL_MODE "function"     
+    CAUSAL_PASS_REGEX
+        "Starting causal experiment #1(.*)causal/experiments.json(.*)causal/experiments.coz"
+    ENVIRONMENT "ROCPROFSYS_VERBOSE=3"
+)
+
+rocprofiler_systems_add_causal_test(
+    SKIP_BASELINE
+    DISABLE_DOUBLE_HYPHEN 
+    NAME causal-no-double-hyphen
+    TARGET causal-cpu-rocprofsys
+    RUN_ARGS 70 10 432525 400000000
+    CAUSAL_MODE "function"
+      CAUSAL_PASS_REGEX
+        "Starting causal experiment #1(.*)causal/experiments.json(.*)causal/experiments.coz"
+    ENVIRONMENT "ROCPROFSYS_VERBOSE=1"
+)
```

```suggestion
add_test(
    NAME causal-no-double-dash
    COMMAND
        $<TARGET_FILE:rocprofiler-systems-causal> $<TARGET_FILE:causal-cpu-rocprofsys>
        "70" "10" "432525" "400000000"
    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
)

set_tests_properties(
    causal-no-double-dash
    PROPERTIES
        ENVIRONMENT
            "ROCPROFSYS_CI=ON;ROCPROFSYS_CI_TIMEOUT=45;ROCPROFSYS_VERBOSE=1;ROCPROFSYS_TIME_OUTPUT=OFF;ROCPROFSYS_USE_PID=OFF;ROCPROFSYS_OUTPUT_PATH=${PROJECT_BINARY_DIR}/rocprof-sys-tests-output/;ROCPROFSYS_OUTPUT_PREFIX=causal-no-double-dash/"
        TIMEOUT 45
        LABELS "causal"
        PASS_REGULAR_EXPRESSION
            "Starting causal experiment #1(.*)causal/experiments.json(.*)causal/experiments.coz"
)
```
I probably wouldn't add a parameter to the "add_causal_test" function just for this. You can just use the standard "add_test" function. Something like this.

---

Thanks for the suggestion. I’ll stick with add_test for this case instead of adding a new parameter to add_causal_test